File: Inventor-Training-Material-0c232f96f36e1e0bed31b07a6a6ebb901a31dd40/Module 08 - DotNet AddIns/Labs Solution - AddIn.VB.Net - registry/AutomationExe.VB.Net/AutomationExe/Form1.vb

Public Class Form1

    Dim mAddInInterface As HelloWorldAddIn.HelloWorldAddIn.MyInterface

    Public Sub New()

        ' This call is required by the Windows Form Designer.
        InitializeComponent()

    End Sub

    Private Sub Form1_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        Try
            Dim oApp As Inventor.Application = System.Runtime.InteropServices.Marshal.GetActiveObject("Inventor.Application")

            Try

                Dim oAddIn As Inventor.ApplicationAddIn = oApp.ApplicationAddIns.ItemById("{958eaca3-2151-415a-aa22-02d5e495389a}")
                mAddInInterface = oAddIn.Automation

            Catch ex As Exception

                System.Windows.Forms.MessageBox.Show("Error: AddIn not found...")
                Button1.Enabled = False
                Exit Sub

            End Try

        Catch ex As Exception

            System.Windows.Forms.MessageBox.Show("Error: Inventor must be running...")
            Button1.Enabled = False
            Exit Sub

        End Try
    End Sub


    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs)

        If (TextBox.Text.Length > 0) Then

            mAddInInterface.MyFunction(TextBox.Text)

        End If

    End Sub

End Class

File: Inventor-Training-Material-0c232f96f36e1e0bed31b07a6a6ebb901a31dd40/Module 08 - DotNet AddIns/Labs Solution - AddIn.CSharp - registry-free/HelloWorldCS/Automation/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;


namespace Automation
{
    public partial class Form1 : Form
    {
        Inventor.Application _InvApplication;
        HelloWorldCS.AutomationInterface _InvAddInInterface; 

        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            try
            {
                _InvApplication = System.Runtime.InteropServices.Marshal.GetActiveObject("Inventor.Application") as Inventor.Application;
            }

            catch
            {
                Type inventorAppType = System.Type.GetTypeFromProgID("Inventor.Application");
                _InvApplication = System.Activator.CreateInstance(inventorAppType) as Inventor.Application;
            }

            _InvApplication.Visible = true;

            string addInCLSID = "{28e1f9bc-44fb-464c-ba98-e9c14c7eed44}";

            Inventor.ApplicationAddIn addIn = _InvApplication.ApplicationAddIns.get_ItemById(addInCLSID.ToUpper());

            //Make sure addin is activated
            if (!addIn.Activated)
            {
                addIn.Activate();
            }

            _InvAddInInterface = addIn.Automation as HelloWorldCS.AutomationInterface;
        }

        private void button1_Click(object sender, EventArgs e)
        {
            _InvAddInInterface.Execute(textBox1.Text);
        }

        private void button2_Click(object sender, EventArgs e)
        {
            string param = _InvAddInInterface.GetParam();
            System.Windows.Forms.MessageBox.Show(param, "Param from AddIn");
        }
    }
}
